dev/com.ibm.ws.microprofile.faulttolerance.cdi/src/com/ibm/ws/microprofile/faulttolerance/cdi/FaultToleranceCDIExtension.java

```diff
@@ -170,4 +173,31 @@ public void beforeBeanDiscovery(@Observes BeforeBeanDiscovery beforeBeanDiscover
         AnnotatedTypeWrapper<T> wrapper = new AnnotatedTypeWrapper<T>(beanManager, processAnnotatedType.getAnnotatedType(), interceptedClass, interceptedMethods);
         processAnnotatedType.setAnnotatedType(wrapper);
     }
+
+    // Move the FaultToleranceInterceptor up before the Transactional interceptors if required
+    public void afterTypeDiscovery(@Observes AfterTypeDiscovery atd) {
+        if (Boolean.getBoolean("com.ibm.ws.faulttolerance.run.before.transactional")) {
```

Comment:
This should be statically cached at the class level rather than being called every time. It should also have `microprofile` in the name ... e.g. `com.ibm.ws.microprofile.faulttolerance.before.transactional`
It seems unlikely but given that this is a microprofile thing, should the config api be used to get this value? Ask Emily.

Reply:
Does not need to be cached. However, config API is now used.